crates/oxc_ast/src/ast/ts.rs

```diff
@@ -826,6 +826,26 @@ pub struct Decorator<'a> {
     pub expression: Expression<'a>,
 }
 
+#[derive(Debug, Serialize, PartialEq, Eq, Hash)]
+#[serde(tag = "type", rename_all = "camelCase")]
+pub enum Modifier {
+    Abstract(Span),
+    Accessor(Span),
+    Async(Span),
+    Const(Span),
+    Declare(Span),
+    Default(Span),
+    Export(Span),
+    In(Span),
+    Public(Span),
+    Private(Span),
+    Protected(Span),
+    Readonly(Span),
+    Static(Span),
+    Out(Span),
+    Override(Span),
+}
```

Comment:
I kind of prefer 

```
struct Modifier {
  span: Span,
  kind: ModifierKind
}
```

So it's slight easier to access `span` without implementing the `GetSpan` trait.

Reply:
Done.